code/modules/research/experimentor/experimentor.dm

```diff
@@ -0,0 +1,322 @@
+	if(stat>60 && stat<81)
+		return "High"
+	if(stat>80)
+		return "Very High"
+
+/obj/machinery/r_n_d/experimentor/attack_hand(mob/user)
+	user.set_machine(src)
+	var/dat = "<center>"
+	if(istype(loaded_item, /obj/item/unknown_tech))
+		var/obj/item/unknown_tech/T = loaded_item
+		dat += "<b>Loaded Item:</b> [T.name]<br>"
+		if(precise_scanner>=2)
+			dat += "<b>Item Type:</b> [T.containedtype]<br>"
+		dat += "<br>"
+		if(precise_scanner>=3)
+			dat += "<b>Stability:</b> [T.stability]<br>"
+			dat += "<b>Potency:</b> [T.potency]<br>"
+			dat += "<b>Innovation:</b> [T.innovation]<br>"
+			dat += "<b>Flexibility:</b> [T.flexibility]<br>"
+		else
+			dat += "<b>Stability:</b> [getvaguestat(T.stability)]<br>"
+			dat += "<b>Potency:</b> [getvaguestat(T.potency)]<br>"
+			dat += "<b>Innovation:</b> [getvaguestat(T.innovation)]<br>"
+			dat += "<b>Flexibility:</b> [getvaguestat(T.flexibility)]<br>"
+
+		if(precise_scanner >= 4)
+			dat += "<br><b>Odds:</b>"
+			dat += "<br>Uncommon: [min(((T.uncommon_weighting*T.innovation/100+1)*T.uncommon_base), 100)]%"
+			dat += "<br>Rare: [min(((T.rare_weighting*T.innovation/100+1)*T.rare_base), 100)]%"
+			dat += "<br>Very Rare: [min(((T.vrare_weighting*T.innovation/100+1)*T.vrare_base), 100)]%"
+			dat += "<br>"
+
+		dat += "<br>Available actions:"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_POKE]'>Tinker</A></b>"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_IRRADIATE];'>Invent</A></b>"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_GAS]'>Conserve</A></b>"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_HEAT]'>Overclock</A></b>"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_COLD]'>Cool</A></b><br>"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_DISCOVER]'>Discover</A></b>"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_OBLITERATE]'>Scramble</A></b><br>"
+		dat += "<br><b><a href='byond://?src=[UID()];function=eject'>Eject</A>"
+	else if(loaded_item != null)
+		dat += "<b>Loaded Item:</b> [loaded_item.name]<br>"
+		dat += "<br>Available actions:"
+		dat += "<br><b><a href='byond://?src=[UID()];item=\ref[loaded_item];function=[SCANTYPE_RECYCLE]'>Recycle</A></b><br>"
+		dat += "<br><b><a href='byond://?src=[UID()];function=eject'>Eject</A>"
+	else
+		dat += "<b>Nothing loaded.</b>"
+	dat += "<br><a href='byond://?src=[UID()];function=refresh'>Refresh</A><br>"
+	dat += "<br><a href='byond://?src=[UID()];close=1'>Close</A><br></center>"
+	var/datum/browser/popup = new(user, "experimentor","Experimentor", 500, 700, src)
+	popup.set_content(dat)
+	popup.open()
+	onclose(user, "experimentor")
+
+/obj/machinery/r_n_d/experimentor/Topic(href, href_list)
+	var/obj/item/unknown_tech/T = locate(href_list["item"]) in src
```

Comment:
this var declaration can be lowered till you are sure it is required (small optimization)